mesonbuild/modules/i18n.py

```diff
@@ -360,11 +361,19 @@ def itstool_join(self, state: 'ModuleState', args: T.List['TYPE_var'], kwargs: '
         command: T.List[T.Union[str, build.BuildTarget, build.CustomTarget,
                                 build.CustomTargetIndex, 'ExternalProgram', mesonlib.File]] = []
         command.extend(state.environment.get_build_command())
+
+        # shlex.join() was added in Python 3.8
+        itstool_cmd = self.tools['itstool'].get_command()
+        if hasattr(shlex, 'join'):
+            itstool_cmd_str = shlex.join(itstool_cmd)
+        else:
+            itstool_cmd_str = ' '.join(shlex.quote(c) for c in itstool_cmd)
```

Comment:
Avoid hasattr. The `else` branch is quite simple and works everywhere so I'd simply use that even on python 3.8, though you could add a `# TODO: python 3.8 can use shlex.join`and we can switch it over sometime in the future.